src/error/metadata/mod.rs

```diff
@@ -136,7 +136,14 @@ impl From<&mut saucer::Error> for Metadata {
                 RoverClientError::GraphQl { .. } => (None, None),
                 RoverClientError::IntrospectionError { .. } => (None, Some(Code::E011)),
                 RoverClientError::ClientError { .. } => (None, Some(Code::E012)),
-                RoverClientError::InvalidKey => (Some(Suggestion::CheckKey), Some(Code::E013)),
+                RoverClientError::InvalidKey => {
+                    let suggestion_key = match std::env::var(RoverEnvKey::Key.to_string()) {
```

Comment:
you may consider passing the value of `rover.get_env_var(RoverEnvKey::Key)` to the `RoverClientError::InvalidKey` variant in order to integrate with the test framework we have for environment variables, [like so](https://github.com/apollographql/rover/blob/cfe2ea0958eece4be3cb54f0855f2624b30d2f03/src/utils/telemetry.rs#L84)

Reply:
`RoverClientError::InvalidKey` usage is located [here](https://github.com/apollographql/rover/blob/4d42c2eb205e9b1c6ccbc662404e3125833a72f4/crates/rover-client/src/operations/config/who_am_i/runner.rs#L64), and I don't know how to root call this env var without doing sort of props drilling.

Reply:
yeah props drilling is what we would need here i think. we do kinda want to remove this type of env parsing though but haven't done it because of backwards compatibility issues. i'm interested in your take on https://github.com/apollographql/rover/issues/1047

Reply:
Put a comment and I suggest to merge this PR as it and consider a better way when working on #1047 WDYT?

Reply:
sounds good to me!